bundle/bundle.go

```diff
@@ -343,13 +402,178 @@ func (r *Reader) Read() (Bundle, error) {
 	return bundle, nil
 }
 
+func (r *Reader) checkSignaturesAndDescriptors(signatures *SignaturesConfig, descriptors []*Descriptor) error {
+	if signatures == nil && r.verificationConfig != nil {
+		msg := "bundle reader provided with signature verification config but .signatures.json file not included in the bundle"
+		return fmt.Errorf("%v", msg)
+	}
+
+	if signatures != nil {
+		if r.verificationConfig == nil {
+			msg := "reader missing signature verification config (hint: check opa configuration includes key(s))"
+			return fmt.Errorf("%v", msg)
+		}
+
+		// verify the JWT signatures included in the `.signatures.json` file
+		if err := r.verifyBundleSignature(signatures); err != nil {
+			return err
+		}
+
+		// check that number of files in the bundle signatures is equal to number of files in the bundle
+		if len(r.files) != len(descriptors) {
```

Comment:
I think it would be better to report (1) missing and (2) extra files in the bundle. Currently we're reporting the # of files but that information is not very useful from the user's perspective (e.g., what are they going to do if they see that 7 != 6? We can just perform a diff and return something much more useful.)

Reply:
Updated error message format: `file(s) [a/b/c/data.json http/policy/policy.rego] specified in bundle signatures but not found in  the target bundle`